shell/platform/android/test/io/flutter/embedding/engine/FlutterEngineCacheTest.java

```diff
@@ -55,4 +55,21 @@ public void itRemovesFlutterEngines() {
     // --- Verify Results ---
     assertNull(cache.get("my_flutter_engine"));
   }
+
+  @Test
+  public void itRemovesAllFlutterEngines() {
+    // --- Test Setup ---
+    FlutterEngine flutterEngine = mock(FlutterEngine.class);
+    FlutterEngine flutterEngine2 = mock(FlutterEngine.class);
+    FlutterEngineCache cache = new FlutterEngineCache();
+
+    // --- Execute Test ---
+    cache.put("my_flutter_engine", flutterEngine);
+    cache.put("my_flutter_engine_2", flutterEngine2);
+    cache.clear();
```

Comment:
assert here that those engines are indeed there and equal to `flutterEngine` and `flutterEngine2` before clearing

Reply:
Thank you for your comment, I've made another commit, mind to re-check?